apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/InAppWalletsPage.tsx

```diff
@@ -88,50 +73,35 @@ const DashboardConnectEmbeddedWallets: ThirdwebNextPage = () => {
             <ApiKeysMenu
               apiKeys={apiKeys}
               selectedKey={selectedKey}
-              onSelect={setSelectedKey}
+              onSelect={(key) => {
+                setSelectedKey(key);
+                // add clientId search param to the url without affecting router
+                window.history.replaceState(
+                  {},
+                  "",
+                  `/dashboard/connect/in-app-wallets?clientId=${key.key}`,
+                );
```

Comment:
I wonder if this *should* affect router and we instead pass only the specific apiKey down every time? why not let the router do the filtering?

Reply:
updated - added clientId in the route